packages/rum/src/userActionCollection.ts

```diff
@@ -65,42 +32,56 @@ export interface AutoUserAction {
 
 export type UserAction = CustomUserAction | AutoUserAction
 
+interface PendingAutoUserAction {
+  id: string
+  startTime: number
+  stop(): void
+}
+export let currentUserAction: PendingAutoUserAction | undefined
+
+export function stopCurrentUserAction() {
+  if (currentUserAction) {
+    currentUserAction.stop()
+  }
+}
+
 export function startUserActionCollection(lifeCycle: LifeCycle) {
+  let stopNewUserAction: { stop(): void } = {
+    stop: noop,
+  }
+  addEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
   function processClick(event: Event) {
     if (!(event.target instanceof Element)) {
       return
     }
-
-    newUserAction(lifeCycle, UserActionType.CLICK, getElementContent(event.target))
+    stopNewUserAction = newUserAction(lifeCycle, UserActionType.CLICK, getElementContent(event.target))
   }
 
-  addEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
-
   return {
     stop() {
+      stopNewUserAction.stop()
       removeEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
     },
   }
 }
 
-let currentUserAction: { id: string; startTime: number } | undefined
-
-function newUserAction(lifeCycle: LifeCycle, type: UserActionType, name: string) {
+function newUserAction(lifeCycle: LifeCycle, type: UserActionType, name: string): { stop(): void } {
   if (currentUserAction) {
-    // Discard any new user action if another one is already occuring.
-    return
+    // Discard any new click user action if another one is already occuring.
+    return { stop: currentUserAction.stop }
   }
 
   const id = generateUUID()
   const startTime = performance.now()
-  currentUserAction = { id, startTime }
 
-  const { observable: pageActivitiesObservable, stop: stopPageActivitiesTracking } = trackPageActivities(lifeCycle)
   const { eventCounts, stop: stopEventCountsTracking } = trackEventCounts(lifeCycle)
 
-  waitUserActionCompletion(pageActivitiesObservable, (endTime) => {
-    stopPageActivitiesTracking()
+  function closeUserAction() {
     stopEventCountsTracking()
+    currentUserAction = undefined
+  }
+
+  const { stop: stopWaitIdlePageActivity } = waitIdlePageActivity(lifeCycle, (endTime) => {
     if (endTime !== undefined) {
```

Comment:
either for page or user action, it is not explicit what has happened when `endTime === undefined`.
What about updating the callback signature to something like`(hadActivity, endTime) => void`?
it would allow us to do:
```
if (hadActivity) {
  // notify user action
}
```
```
const loadingEndTime = hadActivity ? endTime : performance.now()
```
which would be way more declarative and could probably spare the need to add the comment on trackLoadDuration